encodings/fastlanes/src/rle/compress.rs

```diff
@@ -90,18 +94,37 @@ where
 
     unsafe {
         values_buf.set_len(value_count_acc);
-        indices_buf.set_len(array.len().next_multiple_of(FL_CHUNK_SIZE));
+        indices_buf.set_len(padded_len);
     }
 
     RLEArray::try_new(
         values_buf.into_array(),
-        indices_buf.into_array(),
-        value_chunk_offsets.into_array(),
-        array.validity().clone(),
+        PrimitiveArray::new(indices_buf.freeze(), padded_validity(array)).into_array(),
+        values_idx_offsets.into_array(),
         array.len(),
     )
 }
 
+/// Returns validity padded to the next 1024 chunk for a given array.
+fn padded_validity(array: &PrimitiveArray) -> Validity {
+    match array.validity() {
+        Validity::NonNullable => Validity::NonNullable,
+        Validity::AllValid => Validity::AllValid,
+        Validity::AllInvalid => Validity::AllInvalid,
+        Validity::Array(validity_array) => {
+            let padded_length = array.len().next_multiple_of(FL_CHUNK_SIZE);
+            let bool_array = validity_array.to_bool();
+            let bool_buffer = bool_array.boolean_buffer();
+            let padded_validity_iter = bool_buffer
+                .iter()
+                .take(array.len())
+                .chain(std::iter::repeat_n(false, padded_length - array.len()));
+
+            Validity::from_iter(padded_validity_iter)
```

Comment:
from_iter is cute for tests but we can't do that here. Allocate a new buffer, copy existing validity and zero the trailer